SpatialGDK/Source/SpatialGDKServices/Private/LocalReceptionistProxyServerManager.cpp

```diff
@@ -0,0 +1,265 @@
+	if (ExitCode == 0 && bSuccess)
+	{
+		FRegexPattern ProcessNamePattern(TEXT("\"(.+?)\""));
+		FRegexMatcher ProcessNameMatcher(ProcessNamePattern, TaskListResult);
+		if (ProcessNameMatcher.FindNext())
+		{
+			OutProcessName = ProcessNameMatcher.GetCaptureGroup(1 /* Get the Name of the process, which is the first group. */);
+
+			return true;
+		}
+	}
+
+	UE_LOG(LogLocalReceptionistProxyServerManager, Warning, TEXT("Failed to get the name of the process that is blocking the required port."));
+	
+	return false;
+}
+
+bool FLocalReceptionistProxyServerManager::CheckIfPortIsBound(int32 Port, FString& OutPID, FString& OutLogMsg)
+{
+	FString State;
+	FString ProcessName;
+	bool bSuccess = SpatialCommandUtils::GetProcessInfoFromPort(Port, OutPID, State);
+	if (bSuccess)
+	{
+		if (State.Contains("LISTENING"))
+		{
+			if (GetProcessName(OutPID, ProcessName))
+			{
+				OutLogMsg = TEXT("%s process with PID: %s"), ProcessName, OutPID;
+				return true;
+			}
+
+			OutLogMsg = TEXT("Unknown process with PID: %s."), *OutPID;
+
+			return true;
+		}
+	}
+	OutLogMsg = TEXT("No Process is blocking the required port.");
+
+	return false;
+}
+
+bool FLocalReceptionistProxyServerManager::LocalReceptionistProxyServerPreRunChecks(int32 ReceptionistPort)
+{
+	FString OutLogMessage;
+	FString PID;
+
+	// Check if any process is blocking the receptionist port
+	if (CheckIfPortIsBound(ReceptionistPort, PID, OutLogMessage))
+	{
+		// Try killing the process that blocks the receptionist port if the process blocking the port is a previously running proxy.
+		if (PID == ParsePid())
+		{
+			bool bProcessKilled = SpatialCommandUtils::TryKillProcessWithPID(PID);
+			if (!bProcessKilled)
+			{
+				UE_LOG(LogLocalReceptionistProxyServerManager, Warning, TEXT("Failed to kill the process that is blocking the port. %s"), *OutLogMessage);
+				return false;
+			}
+
+			UE_LOG(LogLocalReceptionistProxyServerManager, Log, TEXT("Succesfully killed %s"), *OutLogMessage);
+			return true;
+		}
+
+		UE_LOG(LogLocalReceptionistProxyServerManager, Log, TEXT("The required port is blocked from a different process with Pid: %s"), *PID);
+		return false;
+	}
+
+	UE_LOG(LogLocalReceptionistProxyServerManager, Log, TEXT("The required port is not blocked!"));
+	return true;
+}
+
+void FLocalReceptionistProxyServerManager::Init(int32 Port)
+{
+	if (!IsRunningCommandlet())
+	{
+		LocalReceptionistProxyServerPreRunChecks(Port);
+	}
+}
+
+
+bool FLocalReceptionistProxyServerManager::TryStopReceptionistProxyServer()
+{
+	if (ProxyServerProcHandle.IsValid())
+	{
+		SpatialCommandUtils::StopLocalReceptionistProxyServer(ProxyServerProcHandle);
+		bProxyIsRunning = false;
+		return true;
+	}
+
+	return false;
+}
+
+
+TSharedPtr<FJsonObject> FLocalReceptionistProxyServerManager::ParsePidFile()
+{
+	FString ProxyInfoFileResult;
+	TSharedPtr<FJsonObject> JsonParsedProxyInfoFile;
+
+	if (FFileHelper::LoadFileToString(ProxyInfoFileResult, *ProxyInfoFilePath))
```

Comment:
The file wouldn't exist before you start the proxy for the first time so we shouldn't error if the file isn't there.

Reply:
I am creating the file before I call this function so it is supposed to be there either way